### ISSUE TITLE
[ops/eltwise]: binary_ng reads a width-over-padded operand's padding as data (silent wrong answer, first call)

### ISSUE THREAD

**Issue by dchenTT** [tenstorrent/tt-metal#56707]
## Component / Area

ops — `ttnn.add` / `subtract` / `multiply` (`operations/eltwise/binary_ng`), interleaved no-broadcast path

## Observed

`binary_ng` reads a **width-over-padded** operand's zero padding as data. Every tile-row after the first
is wrong.

This is not a program-cache bug. It is wrong on the **first call** against a cold cache, with no second
call and no cache hit involved.

## Expected

The result matches the torch reference, as it does when the same operand is height-over-padded.

## Steps (exact commands)

Real Wormhole, `origin/main` @ `286aba86ca1`.

```python
import torch, ttnn

dev = ttnn.open_device(device_id=0)
torch.manual_seed(0)
ta = torch.rand([64, 64], dtype=torch.bfloat16)
tb = torch.rand([64, 64], dtype=torch.bfloat16)

# a: logical [64,64] but padded to [64,128] -> aWt = 4
rm = ttnn.from_torch(ta, dtype=ttnn.bfloat16, layout=ttnn.ROW_MAJOR_LAYOUT,
                     device=dev, memory_config=ttnn.DRAM_MEMORY_CONFIG)
a = ttnn.tilize_with_val_padding(rm, [64, 128], 0.0, memory_config=ttnn.DRAM_MEMORY_CONFIG)

b = ttnn.from_torch(tb, dtype=ttnn.bfloat16, layout=ttnn.TILE_LAYOUT,
                    device=dev, memory_config=ttnn.DRAM_MEMORY_CONFIG)

out = ttnn.to_torch(ttnn.add(a, b))          # c: logical [64,64], padded [64,64] -> cWt = 2

print((out - (ta + tb)).abs().max())          # 1.0   <- wrong
print((out[32:64] - tb[32:64]).abs().max())   # 0.0   <- rows 32-63 are exactly 0 + b
```

## Measured evidence

| Check | Result |
|---|---|
| Control: `ttnn.to_torch(a)` vs `ta` — does the over-padded tensor read back? | **exact, 0.000000.** `tilize_with_val_padding` is not at fault |
| `ttnn.add(a_width-over-padded, b)` vs `ta + tb` | **max abs diff 1.000000, 2026 of 4096 elements wrong** |
| ↳ rows 0-31 | 0.007812 — bf16 rounding, correct |
| ↳ rows 32-63 | 1.000000 — wrong |
| Hypothesis: rows 32-63 equal `0 + b`, i.e. `a`'s zero padding | **exact match, 0.000000** |
| Control: **height**-only over-padding (`[64,64]` -> `[128,64]`) | **correct** (0.007812, bf16 rounding) |

The output's second tile-row is bit-exactly `0 + b`. That is the operand's padding, not noise.

## Mechanism

The no-broadcast reader advances `a`'s page pointer by the **output's** `Wt`:

- `.../binary_ng/device/kernels_ng/dataflow/reader_interleaved_no_bcast.cpp` — initial offset
  `start_th * Wt`, then `tile_offset += Wt` per row.
- `.../binary_ng/device/binary_ng_program_factory.cpp` — the reader runtime args pass `cD, cN, cC, cHt,
  cWt`, the **output's** grid, while `a`'s own extents appear only in the outer-dimension strides
  `aHt * aWt * aC * ...`. So `a` has no row stride of its own.

With `aWt = 4` and `cWt = 2` the reader fetches `a` tiles `0,1,2,3` where the correct tiles are
`0,1,4,5`. Tiles `2,3` are the width padding.

`compute_output_specs` gives an interleaved output the default alignment. An over-padded input paired
with a normally-padded output is therefore the ordinary case, not an exotic one.

## Affected architectures

**Wormhole: measured.** The numbers above are from real Wormhole hardware, not a simulator.

**Blackhole: expected, not measured.** The defect is in the production op, which all architectures
share. Quasar is the one target unaffected, because it has a separate copy under
`experimental/quasar/binary_ng/`. The reader kernel has no architecture branching — its only
conditionals are `SRC_SHARDED` / `SRC_SHARDED_B`, which select sharded versus interleaved — and the host
passes `cHt`/`cWt` the same way on every target. So Blackhole should behave identically, but this report
does not claim a Blackhole measurement.

## Why this has gone unnoticed

Height-over-padding is harmless: the extra tile *rows* are never visited. Height is also the direction
the in-tree over-padding call sites produce (`_nearest_32_per_core` in `test_group_norm_DRAM.py`, and the
`tilize_with_val_padding` uses in `test_to_layout.py` and `test_tilize_untilize_2D.py`). I found no
in-tree producer that feeds a width-over-padded tensor to a binary op, so the defect is reachable from
user code but not from CI.

`ttnn.tilize_with_val_padding` takes a caller-supplied padded shape and is public Python, so reaching it
needs no internal API.

## Suggested fix

Give the reader `a`'s and `b`'s own row extent instead of reusing the output's — pass `aWt`/`bWt`
alongside the existing outer strides and use them for the per-row advance.

A `TT_FATAL` rejecting a width-over-padded operand would close the hole, but it would also reject a shape
the op could handle.

Only the `no_bcast` path was examined. The broadcast readers in the same directory should be checked for
the same pattern.

## Frequency

100% — every call with a width-over-padded operand and more than one output tile-row.

## Software Versions

tt-metal `286aba86ca1` (2026-09-15). Also reproduced a week earlier at `e7638d2859b` with identical
numbers; the reader kernel has had no commits in between.

## Hardware Details

Wormhole, real hardware. See "Affected architectures" for Blackhole.

## Is this a regression?

Unknown — not bisected. The reader has carried this stride for at least the range examined.

## Priority

High. It is a silent wrong answer in a core eltwise op, on a shape reachable from public Python.

## Impact

Any model or user code that feeds a width-over-padded tensor into `ttnn.add`, `subtract` or `multiply`
gets wrong numbers with no error and no warning. Related note: the program-cache issues #55414 / #55690
are in the same input space, and any regression test for those must use **height** over-padding, or it
trips this defect and fails for the wrong reason.
